docs/src/lib/indexing.md

```diff
@@ -109,15 +109,12 @@ then view points to selected columns by their number at the moment of creation o
 
 ## `setindex!`
 
-The following list specifies the **target** behavior of `setindex!` operations depending on argument types.
-
-In the current release of DataFrames.jl we are in the transition period when an old, undocumented, behavior
-of `setindex!` is still supported, but throws deprecation warnings.
-
-The behavior described below will be fully implemented in the next major release of DataFrames.jl.
+The following list specifies the behavior of `setindex!` operations depending on argument types.
 
 In particular a description explicitly mentions if the assignment is *in-place*.
 
+Note that `setindex!` operations are not atomic.
```

Comment:
this means that if `setindex!` operates and fails "halfway" it will throw an error and possibly leave garbage in target `DataFrame` (it does not clean-up as `push!` or `append!` now do)

Reply:
I will change the description